9 delete comment

MFStevenson opened the pull request.

nc-poppy-m reviewed (changes requested):
Great work so far! A few changes to consider:

- We'd only want the user to delete comments they've posted. (You can keep the user in state when logged in, or hardcode a user)
- I'd ideally like to see some kind of indication that it's in the process of being deleted, I kept clicking as it took a little while to clear 

Implement these changes, then resubmit in the morning :)

MFStevenson committed:
actions feedback
sets loading state for slow connection
hardcoded user can delete comments

nc-poppy-m commented on src/pages/CommentsPage.jsx:
```diff
@@ -1,27 +1,44 @@
-  const handleBackToArticle = () => {
-    navigate(`/articles/${article_id}`);
-  };
+  }, [handleSubmit]);
```
This might be where your infinite re-render is coming from

nc-poppy-m reviewed (changes requested):
Brilliant implementation of feedback! Perhaps the loading spinner could be a bit smaller or you could create a new component for this kind of thing for UX purposes, but that's something that can be tackled later.

However, I've spotted an infinite re-render in your comments section. I have a feeling that it's to do with how you've added your 'handleSubmit' in your dependency array. Fix this and resubmit when you're ready.

MFStevenson committed:
prevents infinite loop

nc-modou reviewed (commented):
When a comment is deleted, it still appears to be on the page until the page is refreshed, this isn't ideal because it make it look like the delete hasn't worked even though it has. You should look into making the user feedback better by showing this - perhaps record if a comment has been deleted in it's state then replace it with a 'has been deleted message' if it has been?

Also, though not related to this ticket, the comment input box is at the bottom of the page which doesn't make sense from a user experience standpoint as you've got to scroll to the bottom, then back to the top to see your comment

MFStevenson committed:
when comment deleted replaces with message comment was deleted
comment box at bottom but renders at top, comment box moved in ticket 8 to top of comments

nc-verity-gregory reviewed (changes requested):
As a user I have no idea what comments I have access to delete, every comment has a delete button leading me to believe that I can interact with it. I had to click on at least four comments to find which ones I can actually interact with. If you are going to have that as feedback you really should limit the user to which buttons they can and can't see. Conditional rendering is the way to go here.

I'm sure this may have been mentioned, but it bears repeating. Being taken away from an article to view comments that I then have to scroll all the way down to be able to add my own is not the greatest user flow or experience. 

While testing different internet speeds on chrome, the loading feedback is a little misleading to what is actually happening. In an offline mode I have an infinite spinner now that I can _assume_ something has gone wrong but from a user perspective I have no idea what has happened. It would be great to have a short, simple message that 'something has gone wrong'.

Ideally optimistically rendering deleted comments would have a great experience for users on a slow connection as well. In order to complete this ticket, please make the deletion function clearer on what I can delete, and address the offline infinite loading animation for a clearer error message. (To be clear, you have fixed your infinite loop 👍 I'm looking for the animation to not be there if there is a problem)

MFStevenson committed:
Merge branch 'main' into 9-delete-comment

MFStevenson committed:
actions feedback
changes to optimistically render
addresses offline scenario

popatre reviewed (changes requested):
Looks all good

popatre reviewed (approved):
Yeah looks all good to me 👍
